src/Components/Form/Form.js

```diff
@@ -0,0 +1,90 @@
+    humiditySelected: false,
+    twilightSelected: false,
+  }
+
+  const [weatherDetails, setWeatherDetails] = useState(defaultWeatherDetails);
+
+  const handleInputChange = (e) => {
+    const { name, checked } = e.target;
+    setWeatherDetails({
+      ...weatherDetails,
+      [name]: checked,
+    });
+  }
+
+  const handleAllClick = (e) => {
+    if (!e.target.checked) {
+      setWeatherDetails(defaultWeatherDetails);
+    } else {
+      setWeatherDetails({
+        allWeatherSelected: true,
+        temperatureSelected: true,
+        cloudSelected: true,
+        windSelected: true,
+        humiditySelected: true,
+        twilightSelected: true,
+      });
+    }
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    handleFormSubmission({ userCity, uncleCity, weatherDetails });
+  }
+
+  const handleReset = (e) => {
+    e.preventDefault();
+    setUserCity('');
+    setUncleCity('');
+    setWeatherDetails(defaultWeatherDetails);
+    handleFormReset();
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <fieldset>
+        <legend>Select locations</legend>
+        <div className="location-detail-inputs">
+          <label htmlFor="userCity">Your location: </label>
+          <input type="text" name="userCity" id="userCity" onChange={(e) => { setUserCity(e.target.value) }} value={userCity} required list='commonLocations' />
+          <label htmlFor="uncleCity">Uncle location: </label>
+          <input type="text" name="uncleCity" id="uncleCity" value={uncleCity} onChange={(e) => { setUncleCity(e.target.value) }} required list='commonLocations' />
+        </div>
+        <CityDatalist />
+      </fieldset>
+      <fieldset>
+        <legend>Weather details</legend>
+        <div className="weather-detail-inputs">
+          <input type="checkbox" id="allWeatherSelected" name='allWeatherSelected' onChange={handleAllClick} checked={weatherDetails.allWeatherSelected} />
+          <label htmlFor="allWeatherSelected">All</label>
+          <input type="checkbox" name="temperatureSelected" id="temperature" checked={weatherDetails.temperatureSelected} onChange={handleInputChange} />
+          <label htmlFor="temperature">Temperature</label>
+          <input type="checkbox" name="windSelected" id="wind" checked={weatherDetails.windSelected} onChange={handleInputChange} />
+          <label htmlFor="wind">Wind</label>
+          <input type="checkbox" name="humiditySelected" id="humidity" checked={weatherDetails.humiditySelected} onChange={handleInputChange} />
+          <label htmlFor="humidity">Humidity</label>
+          <input type="checkbox" name="cloudSelected" id="cloud" checked={weatherDetails.cloudSelected} onChange={handleInputChange} />
+          <label htmlFor="cloud">Clouds</label>
+          <input type="checkbox" name="twilightSelected" id="twilight" checked={weatherDetails.twilightSelected} onChange={handleInputChange} />
+          <label htmlFor="twilight">Twilight</label>
```

Comment:
I might change this to "Daylight" rather than "Twilight"... when i clicked on it in the app I had no idea what it was going to give me until it said "uncle has x amount less daylight than you"... I wonder if daylight spells it out a little better since that's really the value you are checking - how much daylight we both had?